Dalamud/Plugin/Services/ITextureProvider.cs

```diff
@@ -281,4 +280,12 @@ public interface ITextureProvider
     /// <returns><c>true</c> if supported.</returns>
     /// <remarks><para>This function does not throw exceptions.</para></remarks>
     bool IsDxgiFormatSupportedForCreateFromExistingTextureAsync(int dxgiFormat);
+
+    /// <summary>Converts an existing <see cref="IDalamudTextureWrap"/> instance to a new instance of
+    /// <see cref="FFXIVClientStructs.FFXIV.Client.Graphics.Kernel.Texture"/>.</summary>
+    /// <param name="wrap">Instance of <see cref="IDalamudTextureWrap"/> to convert.</param>
+    /// <param name="leaveWrapOpen">Whether to leave <paramref name="wrap"/> non-disposed when the returned
+    /// <see cref="Task{TResult}"/> completes.</param>
+    /// <returns>Address of the new <see cref="FFXIVClientStructs.FFXIV.Client.Graphics.Kernel.Texture"/>.</returns>
+    nint ConvertToKernelTexture(IDalamudTextureWrap wrap, bool leaveWrapOpen = false);
```

Comment:
I wonder if it would be wise to write a little bit about lifecycle here? How are you supposed to use this, what happens when you call this with texture wraps from shared textures, what are you supposed to do with the texture you're replacing and do you need to do anything to what you get back here when the texture is no longer used by the game?